### ISSUE TITLE
Is it a bug in yolo/train.py?

### ISSUE THREAD

**Issue by gauss-clb** [thtrieu/darkflow#487]
Not every grid will contain an object, so one grid may predict no objects. But in [yolo/train.py](https://github.com/thtrieu/darkflow/blob/master/darkflow/net/yolo/train.py#L66), `tf.reduce_max(iou, [2], True)` will return max value of two bbox when the grid don't contain an object, and `best_box = tf.equal(iou, tf.reduce_max(iou, [2], True))` will have at least one true for two bbox of each grid.

Do you agree with me ?

**Comment by Ridhwanluthra** [comment]
based on https://github.com/tensorflow/tensorflow/issues/3510
i think that tf.reduce_max return +3.4e+38 it wont be equal to iou in tf.equal operation hence we get a false and hence no bbox is selected.
@thtrieu please correct me if i am wrong

**Comment by gauss-clb** [comment]
@Ridhwanluthra No, there is no tensors with a zero-size dimension, if a grid don't contain an object, `_upleft` and `_botright` will be fed by zeros tensors, so `iou` must have value,  `tf.reduce_max` must return value rather than `+3.4e+38`.
